### PR TITLE
Feature/hero comparison attr list

### PR DESCRIPTION
**Pull request by Surxe**
Form a list of non-constant stats that will be displayed by deadlocked.wiki/Hero Comparison, including SustainedDPS newly added stat

### CODE REVIEW

File: src/parser/parsers/heroes.py [hunk 1]
```diff
@@ -84,7 +84,52 @@ def run(self):
...
+
+    def _write_non_constants_stats(self, all_hero_stats):
+        """
+        Writes list of non-constant stats to json file.
+
+        Returns non_constants_stats dict, where elements are true if they are non-constant.
+
+        Non-constant stats are ones that will be displayed
+        on the deadlocked.wiki/Hero_Comparison page, among others in the future.
+        """
```
Comment (hassayag) [inline src/parser/parsers/heroes.py:100]: I think be clearer as to what non-constant means. Eg. `a "non-constant" stat is one that differs between heroes and is therefore suitable to be displayed on the Hero_Comparison page`

**Review by hassayag — APPROVED** [review]
Looks good. I do worry that it requires too much human intervention to get the data out there. Perhaps a future PR could move that array from the lua script to somewhere in this repo. And we can have CI block the merge if the table is missing an element. Then it can categorically never be missed

File: src/parser/parsers/heroes.py [hunk 2]
```diff
@@ -84,7 +84,52 @@ def run(self):
 
                 all_hero_stats[hero_key] = json_utils.sort_dict(hero_stats)
 
-        return all_hero_stats
+        # Write non-constant stats to json file
+        non_constants = self._write_non_constants_stats(all_hero_stats)
+
+        return all_hero_stats, non_constants
+
+    def _write_non_constants_stats(self, all_hero_stats):
```
[the remark this answers is not in the document]
Comment (Surxe) [reply src/parser/parsers/heroes.py]: Fixed! Previous implementation did write a json, had forgotten to rename it.

File: src/parser/parsers/heroes.py (hunk 1 above)
Comment (Surxe) [reply src/parser/parsers/heroes.py:99]: Renamed from "non-constant-stats" to "meaningful-stats"

**Comment by Surxe** [comment]
Noticed that it wasn't factoring in stats that are initially constants, but have scaling on some heroes. Going to add these then re-request review.

**Comment by Surxe** [comment]
> Noticed that it wasn't factoring in stats that are initially constants, but have scaling on some heroes. Going to add these then re-request review.

Complete!

File: src/parser/parsers/heroes.py (hunk 1 above)
Comment (hassayag) [reply src/parser/parsers/heroes.py:98]: Still requires an explanation as to what meaningful or non-constant means

File: output-data/json/hero-non-constants.json
Comment (hassayag) [inline output-data/json/hero-non-constants.json:1]: delete

**Review by hassayag — CHANGES_REQUESTED** [review]
Needs reparsing, looks to be out of date

File: src/parser/parsers/heroes.py (hunk 1 above)
Comment (Surxe) [reply src/parser/parsers/heroes.py:101]: added

**Comment by Surxe** [comment]
Re-merged develop into branch, decompiled most recent patch, and re-parsed. Should be all good to go now.

File: src/output-data/json/ability-data.json
Comment (hassayag) [inline src/output-data/json/ability-data.json:1]: i think this is a remnant from the bad env var. delete

Comment (Surxe) [reply src/output-data/json/ability-data.json:1]: > i think this is a remnant from the bad env var. delete

not sure what you mean, ability-data.json is referenced by parser.py and changelogs.py

Comment (hassayag) [reply src/output-data/json/ability-data.json:1]: The directory is not correct

File: output-datajson/hero-meaningful-stats.json
Comment (hassayag) [inline output-datajson/hero-meaningful-stats.json:1]: another dodgy file path

Comment (Surxe) [reply output-datajson/hero-meaningful-stats.json:1]: nice catch hah, fixed!

File: src/output-data/json/ability-data.json
Comment (Surxe) [reply src/output-data/json/ability-data.json:1]: Ooooh I see what you mean now, okay fixed this as well

**Comment by Surxe** [comment]
Remerged develop branch

File: output-data/version.txt [hunk 3]
```diff
@@ -1,9 +1,9 @@
...
+ServerVersion=5220
 ProductName=citadel
 appID=1422450
 ServerAppID=1422460
 ToolsAppID=211
-SourceRevision=9219272
-VersionDate=Sep 30 2024
-VersionTime=11:12:14
+SourceRevision=9217762
+VersionDate=Sep 27 2024
```
Comment (hassayag) [inline output-data/version.txt:8]: I put the new patch on develop so should reparse and then its good to go!

Comment (Surxe) [reply output-data/version.txt:8]: I already reparsed, looks like it matches develop

Comment (hassayag) [reply output-data/version.txt:8]: this is a diff between your branch and develop. Maybe steam needs to update the game?

**Review by hassayag — APPROVED** [review]
🚀

**Pull request merged**